### PR TITLE
Client Tokens Generic as per app name

### PR DESCRIPTION
**Pull request by Mangoman3**
- old clientTokens will be migrated to newFromat
- newFormat is ProductName:BundleID:PersonalAccessToken

### CODE REVIEW

File: Sources/Base/OAuth2KeychainAccount.swift [hunk 1]
```diff
@@ -59,6 +59,20 @@ struct OAuth2KeychainAccount: KeychainGenericPasswordType {
 		accessGroup = oauth2.keychainAccessGroup
 		data = inData
 	}
+	
+	/**
+	Designated initializer.
+	- parameter _serviceName:  The OAuth2 instance Keychain Entry Name
+	*/
+	
+	init(oauth2: OAuth2Securable, account: String, data inData: [String: Any] = [:],_serviceName:String) {
```
Comment (faizan-ahmed-pure-vpn) [inline Sources/Base/OAuth2KeychainAccount.swift:68]: What we will be passing in _serviceName params?

File: Sources/Base/OAuth2Securable.swift [hunk 2]
```diff
@@ -106,6 +113,20 @@ open class OAuth2Securable: OAuth2Requestable {
 		return "http://localhost"
 	}
 	
+	/** Migrate client credentials from old token to Generic Product Name Format */
+	private func migrateToken(_ toks: OAuth2KeychainAccount, _ toks_data: [String : Any]) throws {
```
Comment (faizan-ahmed-pure-vpn) [inline Sources/Base/OAuth2Securable.swift]: this method name should be named as `migrateTokenAndUpdateInKeyChain` or `migrateTokenAndSaveInKeyChain`

Comment (Mangoman3) [reply Sources/Base/OAuth2Securable.swift]: name is kept similar to methods used as forgetTokens, as we have separate methods for keychain for saving, removing and updating.

File: Sources/Base/OAuth2KeychainAccount.swift (hunk 1 above)
Comment (Mangoman3) [reply Sources/Base/OAuth2KeychainAccount.swift:68]: `/**
	Designated initializer.
	- parameter _serviceName:  The OAuth2 instance Keychain Entry Name
	*/`

**Pull request merged**